pta-intelligent-search-document-processor/src/main/java/fi/maanmittauslaitos/pta/search/documentprocessor/DocumentProcessorFactory.java

```diff
@@ -66,7 +65,9 @@ public Document processDocument(InputStream is) throws DocumentProcessingExcepti
 			org.w3c.dom.Document doc;
 			
 			try {
-				doc = builder.parse(is);
+				synchronized (builder) {
```

Comment:
Did you test if it was necessary to synchronize the entire processing? I know maui needs to be synchronized, but that could be achived by synchronizing only the maui processor

Reply:
Oops. This is just XML parsing synchronization. All good.